Lesson9/src/shop/Shop.java

```diff
@@ -0,0 +1,59 @@
+package shop;
+
+import exceptions.SearchProductException;
+import interfaces.IShop;
+import products.Product;
+
+import java.util.*;
+
+public class Shop implements IShop {
+    List<Product> productList = new ArrayList<>();
+
+    @Override
+    public void addProduct(Product product) throws SearchProductException {
+        if (ShopUtil.isTheProductPresent(productList, product.getId())) {
+            throw new SearchProductException(product.getName() + " is already in the list");
+        }
+        productList.add(product);
+    }
+
+    public void addProduct(List<Product> newList) throws SearchProductException {
+        for (Product product : productList) {
+            for (Product newProduct : newList) {
+                if (newProduct.getId() == (product.getId())) {
+                    throw new SearchProductException(newProduct.getName() + " is already in the list");
+                }
+            }
+        }
+        productList.addAll(newList);
+    }
+
+    @Override
+    public List<Product> getAllProducts() {
+        productList.sort(Comparator.comparingInt(Product::getPrice));
+        return productList;
+    }
+
+    public List<Product> getList() {
+        return productList;
+    }
+
+    @Override
+    public boolean deleteProduct(int id) throws SearchProductException {
+        if (ShopUtil.searchProductByID(productList, id) == null) {
+            throw new SearchProductException("The Product with this ID:" + id + " was not found");
+        } else {
+            Product product = ShopUtil.searchProductByID(productList, id);
+            productList.remove(product);
+            return true;
+        }
```

Comment:
refactor it to look like
 Product product = ShopUtil.searchProductByID(productList, id);
if (product == null) {
            throw new SearchProductException("The Product with this ID:" + id + " was not found");
        } else {
            productList.remove(product);
            return true;
        }
        
        so you don't search for the same id twice